test/e2e/image_pull_error_test.go

```diff
@@ -52,13 +52,10 @@ func TestImagePullError(t *testing.T) {
 		cond := r.Status.GetCondition(v1.ConfigurationConditionReady)
 		if cond != nil && !cond.IsUnknown() {
 			if cond.IsFalse() {
-				if cond.Reason == wantCfgReason {
+				if cond.Reason == wantCfgReason && strings.Contains(cond.Message, "Back-off pulling image") {
```

Comment:
Previously the configuration will be go from status:

```
k get configuration  -n serving-tests 
NAME                        LATESTCREATED                     LATESTREADY   READY     REASON
image-pull-error-dagmxojy   image-pull-error-dagmxojy-00001                 Unknown 
```
to status failed after the progressdeadline was exceeded (120s in tests).
Here instead since the patch as soon as it sees no availability at the deployment side it will mark the revision as ready=false and configuration will get:
```
                    {
                        "lastTransitionTime": "2024-01-25T13:44:18Z",
                        "message": "Revision \"image-pull-error-kbfvvcsg-00001\" failed with message: Deployment does not have minimum availability..",
                        "reason": "RevisionFailed",
                        "status": "False",
                        "type": "Ready"
                    }
```
Later on after progressdeadline is passed it will get expected msg here.
That is why we will have to wait.
```
                "conditions": [
                    {
                        "lastTransitionTime": "2024-01-25T13:36:08Z",
                        "message": "Revision \"image-pull-error-gnwactac-00001\" failed with message: Deployment does not have minimum availability..",
                        "reason": "RevisionFailed",
                        "status": "False",
                        "type": "Ready"
                    }
```

Reply:
Need to cover ```rpc error: code = NotFound desc = failed to pull and unpack image ``` as well, as seen from the failures.